app/interactors/unverified_auth_organizer.rb

```diff
@@ -0,0 +1,35 @@
+class UnverifiedAuthOrganizer
+  attr_reader :auth
+  private :auth
+
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def user
+    if user_found_by_email.present?
+      fail AuthVerificationPolicy::OauthError, "Please, connect your account from profile page."
+    else
+      found_user.send_confirmation_instructions unless found_user.confirmed?
+      found_user
+    end
+  end
```

Comment:
just don't really like if - else thingies, when they can be split into methods

def user
  ensure_user_existance!

  found_user.send_confirmation_instructions unless found_user.confirmed?
  found_user
end

def ensure_user_existance
  unless user_found_by_email
    fail AuthVerificationPolicy::OauthError, "Please, connect your account from profile page."
  end
end